MCprep_addon/materials/generate.py

```diff
@@ -958,6 +958,32 @@ def set_saturation_material(mat):
 		sat_node.mute = not bool(saturate)
 		sat_node.hide = not bool(saturate)
 
+def create_node(tree_nodes, node_type, **kwargs):
+	"""Function to create node"""
+	if node_type == 'ShaderNodeMixRGB': # For MixRGB in 3.4 become Legacy 
+		if util.min_bv((3, 4, 0)):
+			node = tree_nodes.new('ShaderNodeMix')
+			node.data_type = 'RGBA'
+		else:
+			node = tree_nodes.new('ShaderNodeMixRGB')
+	else:
+		node = tree_nodes.new(node_type) 
+	for attr, value in kwargs.items():
+		if hasattr(node, attr):
+			setattr(node, attr, value)
+	return node
+	
+def get_sockets_node(node, is_input=True):
+	n_type = node.bl_idname
+	if n_type == 'ShaderNodeMix' or n_type == 'ShaderNodeMixRGB':
+		if util.min_bv((3, 4, 0)): # For MixRGB in 3.4 become Legacy 
+			if node.data_type == 'RGBA': 
+				inputs, outputs = [0,6,7], [2]
```

Comment:
I'm not sure where these numbers come from. Perhaps add a comment explaining where the numbers come from?